build.gradle

```diff
@@ -236,3 +237,38 @@ allprojects {
         options.addStringOption('Xdoclint:none', '-quiet')
     }
 }
+
+subprojects { subproject ->
+    apply plugin: 'groovy'
+    apply plugin: 'com.diffplug.spotless'
+
+    spotless {
+        java {
+            googleJavaFormat()  // TODO: Pin version here
+            target '**/*.java'
+            targetExclude '**/non_formatted_code/*'
+            importOrder() // Match the import order configuration
+            eclipse().configFile("${rootDir}/tools/gradle/codestyle/java-google-style.xml")
+            removeUnusedImports()
+            trimTrailingWhitespace()
+            endWithNewline()
+        }
+        groovy {
+            target '**/*.gradle'
+            importOrder()
+            trimTrailingWhitespace()
+            endWithNewline()
+        }
+        kotlin {
+            ktlint()  // TODO: Pin version here
+            target '**/*.kt'
+            targetExclude '**/non_formatted_code/**/*'
+            ktfmt("0.39")
+        }
+        // format 'misc', {
+        //     target '**/*.gradle', '**/*.md', '**/*.yaml', '**/*.yml'
+        //     prettier().config(['tabWidth': 4, 'useTabs': false])
+        //     targetExclude '**/generated/**/*'
+        // }
+    }
+}
```

Comment:
How do you get this to be fast? I've tried this and found it to be _painfully_ slow and gave up. Which is why, in the past, the spotless plugin was applied only at the root project and targeted all `**/*.java` files, but that wasn't a panacea either.